test/h2_client_adapter/chatterbox_test.exs

```diff
@@ -207,4 +207,13 @@ defmodule H2ClientAdapter.ChatterboxTest do
   defp pid(string) when is_binary(string) do
     :erlang.list_to_pid('<#{string}>')
   end
+
+  test "Worker stays alive after unlinking" do
+    with_mock :h2_client,
+      start_link: fn _, _, _, _ -> {:ok, spawn_link(fn -> Process.sleep(100000) end)} end do
+        {:ok, conn_ref} = H2Adapter.open("my.domain.at.domain", 1234)
+        Process.exit(conn_ref, :dunno)
+        assert Process.alive?(self())
```

Comment:
We should wait until the `conn_ref` actually exits (monitor?) before we decide on anything. This test can easily pass by chance (die after assert).

Reply:
Is the way it's done now ok?

Reply:
Yeah, it's better :) Still, a bit weird because this assert can never fail - `Process.alive?(self())` cannot return false, a the process would be there to execute this statement, but the test would crash, so I guess let's roll with it.